astrolabe/runner.py

```diff
@@ -223,17 +267,63 @@ def run(self, persist_cluster=False, startup_time=1):
                 self.cluster_name))
 
         return junit_test
+        
+    def wait_for_idle(self):
+        selector = BooleanCallablePoller(
+            frequency=self.config.polling_frequency,
+            timeout=self.config.polling_timeout)
+        LOGGER.info("Waiting for cluster maintenance to complete")
+        selector.poll([self], attribute="is_cluster_state", args=("IDLE",),
+                      kwargs={})
+                      
+    def retrieve_logs(self):
+        data = self.admin_client.request('GET', '/api/private/nds/groups/%s/clusters/%s' % (self.project.id, self.cluster_name)).data
+        
+        if data['clusterType'] == 'SHARDED':
+            rtype = 'CLUSTER'
+            rname = data['deploymentItemName']
+        else:
+            rtype = 'REPLICASET'
+            rname = data['deploymentItemName']
+            
+        params = dict(
+            resourceName=rname,
+            resourceType=rtype,
+            redacted=True,
+            logTypes=['FTDC','MONGODB'],#,'AUTOMATION_AGENT','MONITORING_AGENT','BACKUP_AGENT'],
+            sizeRequestedPerFileBytes=100000000,
+        )
+        data = self.admin_client.request('POST', 'groups/%s/logCollectionJobs' % (self.project.id,), **params).data
```

Comment:
The `AtlasClient` object exposes a fluent API that you should use instead of issuing a raw request here. In this instance, this would look like:

```
data = self.admin_client.groups[self.project.id].logCollectionJobs.post(**params).data
```

Reply:
Changed all instances to this style except for the log downloading one where the URL is provided by the API.